ambry-router/src/main/java/com.github.ambry.router/CloudRouterFactory.java

```diff
@@ -0,0 +1,163 @@
+      throw new IllegalStateException(
+          "Router datacenter " + routerConfig.routerDatacenterName + " is not part of the clustermap");
+    }
+    this.clusterMap = clusterMap;
+    this.notificationSystem = notificationSystem;
+    this.accountService = accountService;
+    MetricRegistry registry = clusterMap.getMetricRegistry();
+    routerMetrics = new NonBlockingRouterMetrics(clusterMap, routerConfig);
+    time = SystemTime.getInstance();
+    KeyManagementServiceFactory kmsFactory =
+        Utils.getObj(routerConfig.routerKeyManagementServiceFactory, verifiableProperties,
+            new ClusterMapConfig(verifiableProperties).clusterMapClusterName, registry);
+    kms = kmsFactory.getKeyManagementService();
+    CryptoServiceFactory cryptoServiceFactory =
+        Utils.getObj(routerConfig.routerCryptoServiceFactory, verifiableProperties, registry);
+    cryptoService = cryptoServiceFactory.getCryptoService();
+    cryptoJobHandler = new CryptoJobHandler(routerConfig.routerCryptoJobsWorkerCount);
+    defaultPartitionClass = clusterMapConfig.clusterMapDefaultPartitionClass;
+
+    // TODO: need to shut pool down on exit, may need to move to RestServer
+    // Or pass it to router
+    requestHandlerPool = getRequestHandlerPool(verifiableProperties, clusterMap, notificationSystem);
+    networkClientFactory = new LocalNetworkClientFactory((LocalRequestResponseChannel) requestHandlerPool.getChannel(),
+        new NetworkConfig(verifiableProperties), new NetworkMetrics(registry), time);
+    logger.info("Instantiated CloudRouterFactory");
+  }
+
+  public static RequestHandlerPool getRequestHandlerPool(VerifiableProperties verifiableProperties,
+      ClusterMap clusterMap, NotificationSystem notificationSystem) throws Exception {
+    ClusterMapConfig clusterMapConfig = new ClusterMapConfig(verifiableProperties);
+    CloudConfig cloudConfig = new CloudConfig(verifiableProperties);
+    // TODO: move properties into maybe routerConfig, frontendConfig
+    ServerConfig serverConfig = new ServerConfig(verifiableProperties);
+    StoreConfig storeConfig = new StoreConfig(verifiableProperties);
+    MetricRegistry registry = clusterMap.getMetricRegistry();
+
+    DataNodeId nodeId = new CloudDataNode(cloudConfig, clusterMapConfig);
+    CloudDestinationFactory cloudDestinationFactory =
+        Utils.getObj(cloudConfig.cloudDestinationFactoryClass, verifiableProperties, registry);
+    CloudDestination cloudDestination = cloudDestinationFactory.getCloudDestination();
+    VcrMetrics vcrMetrics = new VcrMetrics(registry);
+    StoreManager cloudStorageManager =
+        new CloudStorageManager(verifiableProperties, vcrMetrics, cloudDestination, clusterMap);
+    LocalRequestResponseChannel channel = new LocalRequestResponseChannel();
+    ServerMetrics serverMetrics = new ServerMetrics(registry, AmbryRequests.class);
+    StoreKeyFactory storeKeyFactory = Utils.getObj(storeConfig.storeKeyFactory, clusterMap);
```

Comment:
since there are other assumptions in the router that blobids will be used, you can just use BlobIdFactory directly w/o reflection.

Reply:
ok